components/ProfileElements/UserDetailsModal.jsx

```diff
@@ -38,8 +42,24 @@ export default function UserDetailsModal(props) {
                         initialValue={initialFirstName}
                         onChange={setFirstName}
                         onSubmit={() => {
-                            set(ref(db, "users/" + auth.currentUser.uid + "/firstName"), firstName);
-                            setInitialFirstName(firstName);
+                            try {
```

Comment:
can probably pull this out into a setName function and pass "first" or "last" since the first and last name blocks of code are basically the same

Reply:
Yeah cleaner for sure good spot